.env.example

```diff
@@ -1,11 +1,30 @@
 # This file is useful for doing local development
-# when needing to load the postgres database secrets.
+# when needing to load the postgres database secrets and start a locally running
+# pizza oven service
 
+# Database env vars
 DATABASE_PORT=9999
 DATABASE_HOST=localhost
 DATABASE_USER=opensauced-admin
 DATABASE_PASSWORD={YOUR-SECRET-PASSWORD-HERE}
 DATABASE_DBNAME=pizza
 
-# The port for the Pizza oven server
+# The port for the Pizza oven server to use
 SERVER_PORT=8080
+
+# The git provider to use for the pizza oven service.
+# Must be one of "cache" or "memory" to designate the git provider that will be
+# used to clone and access repos.
+# - The "cache" git provider uses a local cache on disk to clone git repos into.
+#   This uses much less memory than in-memory cloning.
+GIT_PROVIDER=cache
```

Comment:
Q: Why provide memory as an option with it's known limitations?

Reply:
I think an in-memory git provider still has some use cases: I can see the `pizza-cli` using the `GitProvider` interface and utilizing the in-memory provider to prevent eating up storage on a users laptop if the CLI needed to clone repos.

Additionally, according to some of the go-git chatter, cloning into memory is _slightly_ more performant than cloning onto a non-volatile memory medium. I believe that is partly from their optimization work abstracting the filesystem away behind an[ interface in the go-billy project](https://github.com/go-git/go-billy).

This could lead to some interesting cases where we deploy a fleet of instances that are running with _both_ in-memory and cache based `GitProvider`s.

People running this service locally or on a very small device (like an edge device) may also wish to preserve disk space. The memory/disk trade off gets really tricky when you start talking about edge devices but I guess I'm not sure the use cases of this running at a networks edge might be. I'm still not _super_ certain what the performance needs for this system will be, so maybe it's overkill to have both of these providers implement the `GitProvider` interface but it at least is a good starting point for anyone who wants to implement their own `GitProvider`.